src/main/scala/edu/berkeley/ce/sparkrocks/Block.scala

```diff
@@ -486,12 +484,14 @@ case class Block(center: Array[Double], faces: Seq[Face], generation: Int=0) ext
     * This function should only be called once all redundant faces have been removed.
     */
   def calcVertices: Map[Face, Seq[Array[Double]]] = {
-    faces.zip (
-      faces map { f1 =>
+    val relevantFaces = nonRedundantFaces
```

Comment:
Is there a reason for this new `val` definition? Maybe it's left over from your old code where you were filtering by the number of vertices.